### PR TITLE
Cleanup code by relying on the existence of resource mappings

### PR DESCRIPTION
**Pull request by nhaehnle**
We've decided a while ago that we no longer want to support compiling without resource mappings, given where we expect APIs to be going. So let's remove some related code to make what's left easier to understand and maintain. This PR is a first big step in this direction, removing a bunch of obsolete code and fixing up (or removing) tests.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit daa41c1882ba772bf2e6f288a28da9ae2415543f
<details><summary>CTS tests (Failed: 0/211879)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35146/70631 (49.8%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35485/70631 (50.2%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39025/70632 (55.3%)</li>
<li>Failed: 0/70632 (0.0%)</li>
<li>Not Supported: 31607/70632 (44.7%)</li><li>Warnings: 0/70632 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39057/70616 (55.3%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31559/70616 (44.7%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchEntryPointMutate.cpp [hunk 1]
```diff
@@ -1825,21 +1769,10 @@ void PatchEntryPointMutate::addUserDataArgs(SmallVectorImpl<UserDataArg> &userDa
...
-      if (m_pipelineState->getShaderOptions(m_shaderStage).updateDescInElf && m_shaderStage == ShaderStageFragment) {
-        // Put set number to register first, will update offset after merge ELFs
-        // For partial pipeline compile, only fragment shader needs to adjust offset of root descriptor.
-        // This is part of the original "partial pipeline compile" scheme, and it uses a magic number for the
-        // PAL metadata register value because the code to fix it up in llpcElfWriter.cpp just fixes up any
-        // register with the magic value, and hopes it lucks out by not getting a false positive.
-        // TODO: Remove all that code once the new "shader/part-pipeline compile" scheme can replace it.
-        static const unsigned DescRelocMagic = 0xA5A5A500;
-        userDataValue = DescRelocMagic | node.innerTable[0].set;
-      }
```
Comment (jasilvanus) [inline lgc/patch/PatchEntryPointMutate.cpp:1856]: Should the whole `DescRelocMagic` thing be removed now, as it seems this was the only place emitting it?

See https://github.com/search?q=repo%3AGPUOpen-Drivers%2Fllpc+descrelocmagic&type=code

Comment (nhaehnle) [reply lgc/patch/PatchEntryPointMutate.cpp:1856]: Yes, makes sense. I will add that.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 9cd39454453ddb8ead83839083cc43edc97e8255
<details><summary>CTS tests (Failed: 0/211876)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35496/70631 (50.3%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35135/70631 (49.7%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39379/70629 (55.8%)</li>
<li>Failed: 0/70629 (0.0%)</li>
<li>Not Supported: 31250/70629 (44.2%)</li><li>Warnings: 0/70629 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39413/70616 (55.8%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31203/70616 (44.2%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

**Review by jasilvanus — DISMISSED** [review]
Thanks, LGTM.

**Comment by nhaehnle** [comment]
ping @amdrexu

**Review by amdrexu — APPROVED** [review]

**Comment by amdrexu** [comment]
Sorry, my PR changed a lot of LIT tests: https://github.com/GPUOpen-Drivers/llpc/pull/2624. You maybe need a rebase to update this PR.

**Comment by amdvlk-admin** [comment]
### Test summary for commit b80e184354df133d250a3d9045af4fbf105c8ead
<details><summary>CTS tests (Failed: 0/211876)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35496/70631 (50.3%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35135/70631 (49.7%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39379/70629 (55.8%)</li>
<li>Failed: 0/70629 (0.0%)</li>
<li>Not Supported: 31250/70629 (44.2%)</li><li>Warnings: 0/70629 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39413/70616 (55.8%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31203/70616 (44.2%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 9a4aff1941db3f46223cd73dec8d20e9bbbe031a
<details><summary>CTS tests (Failed: 0/211876)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35496/70631 (50.3%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35135/70631 (49.7%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39379/70629 (55.8%)</li>
<li>Failed: 0/70629 (0.0%)</li>
<li>Not Supported: 31250/70629 (44.2%)</li><li>Warnings: 0/70629 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39413/70616 (55.8%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31203/70616 (44.2%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

**Review by amdrexu — APPROVED** [review]

**Pull request merged**